### PR TITLE
Automatically detect proper value for flags.tty if not explicitly set

### PR DESCRIPTION
**Pull request by TylerAldrich**
Thanks to the beauty of StackOverflow, I found an easy way to run `architect exec` with no tty: https://superuser.com/a/1430883

I used the following test script to run exec:
```sh
#!/bin/sh
XDG_CONFIG_HOME=~/.config/architect/local/architect /Users/tyler/code/architect/architect-cli/bin/dev exec -a tyler-aldrich -e example-environment -- "sh -c \"ls -la && echo 'hi'\""
```

Running this on this branch now gives the following output:
```
~/code/architect/architect-cli (detect-notty-automatically*) » true | (setsid ./test-script.sh) 2>&1                                                                                  
stdin does not support tty and was automatically disabled, --no-tty should be used
total 68
drwxr-xr-x    1 root     root           116 Aug  4 18:33 .
drwxr-xr-x    1 root     root            17 Jul 20 16:49 ..
-rw-r--r--    1 root     root            15 Jul  5 16:58 .gitignore
-rw-r--r--    1 root     root           107 Jul  5 16:58 Dockerfile
-rw-r--r--    1 root     root          1901 Jul  5 16:58 README.md
-rw-r--r--    1 root     root           537 Aug  4 18:08 architect.yml
-rw-r--r--    1 root     root           341 Aug  4 15:36 index.js
drwxr-xr-x   59 root     root          4096 Jul 20 16:50 node_modules
-rw-r--r--    1 root     root         39745 Jul 20 16:50 package-lock.json
-rw-r--r--    1 root     root           291 Jul  5 16:58 package.json
hi
```

If you add the --no-tty flag, the warning goes away:
```
~/code/architect/architect-cli (detect-notty-automatically*) » true | (setsid ./test-script.sh) 2>&1                                                                                 
total 68
drwxr-xr-x    1 root     root           116 Aug  4 18:33 .
drwxr-xr-x    1 root     root            17 Jul 20 16:49 ..
-rw-r--r--    1 root     root            15 Jul  5 16:58 .gitignore
-rw-r--r--    1 root     root           107 Jul  5 16:58 Dockerfile
-rw-r--r--    1 root     root          1901 Jul  5 16:58 README.md
-rw-r--r--    1 root     root           537 Aug  4 18:08 architect.yml
-rw-r--r--    1 root     root           341 Aug  4 15:36 index.js
drwxr-xr-x   59 root     root          4096 Jul 20 16:50 node_modules
-rw-r--r--    1 root     root         39745 Jul 20 16:50 package-lock.json
-rw-r--r--    1 root     root           291 Jul  5 16:58 package.json
hi
```

Running with `setsid` and `--tty` explicitly passed fails as expected:
```
~/code/architect/architect-cli (detect-notty-automatically*) » true | (setsid ./test-script.sh) 2>&1                                                                                  
stdin does not support tty
Error: stdin does not support tty
    at Exec.run (/Users/tyler/code/architect/architect-cli/src/commands/exec.ts:259:13)
    at async Exec._run (/Users/tyler/code/architect/architect-cli/node_modules/@oclif/core/lib/command.js:67:22)
    at async Config.runCommand (/Users/tyler/code/architect/architect-cli/node_modules/@oclif/core/lib/config/config.js:268:25)
    at async Object.run (/Users/tyler/code/architect/architect-cli/node_modules/@oclif/core/lib/main.js:73:5)
```

Running this command without `setsid` (and therefore with a tty) works as expected:
```
~/code/architect/architect-cli (detect-notty-automatically*) » ./test-script.sh                                                                                                       tyler@tylers-mbp
total 68
drwxr-xr-x    1 root     root           116 Aug  4 18:33 .
drwxr-xr-x    1 root     root            17 Jul 20 16:49 ..
-rw-r--r--    1 root     root            15 Jul  5 16:58 .gitignore
-rw-r--r--    1 root     root           107 Jul  5 16:58 Dockerfile
-rw-r--r--    1 root     root          1901 Jul  5 16:58 README.md
-rw-r--r--    1 root     root           537 Aug  4 18:08 architect.yml
-rw-r--r--    1 root     root           341 Aug  4 15:36 index.js
drwxr-xr-x   59 root     root          4096 Jul 20 16:50 node_modules
-rw-r--r--    1 root     root         39745 Jul 20 16:50 package-lock.json
-rw-r--r--    1 root     root           291 Jul  5 16:58 package.json
hi
```

### CODE REVIEW

File: src/commands/exec.ts [hunk 1]
```diff
@@ -58,20 +70,17 @@ export default class Exec extends BaseCommand {
...
 
     await new Promise((resolve, reject) => {
       const duplex = createWebSocketStream(ws, { encoding: 'utf-8' });
       duplex.pipe(this.getOutputTransform());
 
       if (flags.stdin) {
-        if (flags.tty && !process.stdin.isTTY) {
-          throw new ArchitectError('stdin does not support tty');
+        if (flags.tty) {
+          process.stdin.setRawMode(true);
```
Comment (TylerAldrich) [inline src/commands/exec.ts:82]: This was an existing bug that's now fixed, `process.stdin.setRawMode` is only available if stdin is a TTY: https://nodejs.org/api/tty.html#readstreamsetrawmodemode (it's only part of the `tty.ReadStream` class)

Comment (tjhiggins) [reply src/commands/exec.ts:82]: Nice find. Could you add this as a comment in the code?

File: src/commands/exec.ts [hunk 2]
```diff
@@ -240,24 +245,36 @@ export default class Exec extends BaseCommand {
...
-    const { flags } = await this.parse(Exec);
+    const { args, flags } = await this.parse(Exec);
+
+    // Automatically set tty if the user doesn't supply it based on whether stdin is TTY.
+    if (flags.tty === undefined) {
+      if (process.stdin.isTTY) {
+        flags.tty = true;
+      } else {
+        flags.tty = false;
+        this.log(chalk.yellow('stdin does not support tty and was automatically disabled, --no-tty should be used'));
```
Comment (tjhiggins) [inline src/commands/exec.ts:256]: Hmm I know I said this should be a warning, but if we detect the stdin doesn't support tty then do they really need to specify `--no-tty`?

File: src/commands/exec.ts [hunk 3]
```diff
@@ -34,7 +46,7 @@ export default class Exec extends BaseCommand {
         description: 'Stdin is a TTY.',
```
Comment (tjhiggins) [inline src/commands/exec.ts:46]: Should we include that the default is we auto-detect support?

**Review by tjhiggins — CHANGES_REQUESTED** [review]

**Pull request merged**